File: wasm/examples/nodejs/javascript/general/mnemonic.js
const kaspa = require('../../../../nodejs/kaspa');
const {
    Mnemonic,
} = kaspa;

kaspa.initConsolePanicHook();

(async () => {

    const mnemonic1 = Mnemonic.random();
    console.log("mnemonic1:", mnemonic1);

    const mnemonic2 = new Mnemonic(mnemonic1.phrase);
    console.log("mnemonic2:", mnemonic2);

    // create a seed with a recovery password ("25th word")
    const seed1 = mnemonic1.toSeed("my_password");
    console.log("seed1:", seed1);

    const seed2 = mnemonic2.toSeed("my_password");
    console.log("seed2:", seed2);

    if (seed1 !== seed2) {
        throw Error("mnemonic restore failure");
    } else {
        console.log("mnemonic restore success");
    }

    // create a seed without a recovery password
    const seed3 = mnemonic1.toSeed();
    console.log("seed3 (no recovery password):", seed3);

})();

File: python/examples/mnemonic.py
from kaspa import Language, Mnemonic

if __name__ == "__main__":
    mnemonic1 = Mnemonic.random()
    print(f'mnemonic 1: {mnemonic1.phrase}')

    mnemonic2 = Mnemonic(phrase=mnemonic1.phrase)
    print(f'mnemonic 2: {mnemonic2.phrase}')

    # Create seed with a recovery password (25th word)
    seed1 = mnemonic1.to_seed("my_password")
    print(f'seed1: {seed1}')

    seed2 = mnemonic2.to_seed("my_password")
    print(f'seed2: {seed2}')

    # Create seed without recovery password
    seed3 = mnemonic1.to_seed()
    print(f'seed3 (no recovery password): {seed3}')